libbindgen/src/ir/item.rs

```diff
@@ -1214,8 +1221,45 @@ impl ItemCanonicalPath for Item {
             debug_assert!(is_alias, "How can this ever happen?");
             parent_path.pop().unwrap();
         }
-        parent_path.push(self.real_canonical_name(ctx, true, false));
+        parent_path.push(self.name().with_namespaces().get(ctx));
 
         parent_path
     }
 }
+
+/// Builder struct for naming variations, which hold inside different
+/// flags for naming options.
+#[derive(Debug)]
+pub struct NameOptions<'a> {
+    item: &'a Item,
+    with_namespaces: bool,
+    for_name_checking: bool,
+}
+
+impl<'a> NameOptions<'a> {
+    /// Construct a new `NameOptions`
+    pub fn new(item: &Item) -> NameOptions {
+        NameOptions {
+            item: item,
+            with_namespaces: false,
+            for_name_checking: false,
+        }
+    }
+
+    /// Set `with_namespaces` flag to `true`
+    pub fn with_namespaces(&mut self) -> &'a mut NameOptions {
+        self.with_namespaces = true;
+        self
+    }
+
+    /// Set `for_name_checking` flag to `true`
```

Comment:
Nitpick: These documentation comments describe the implementation mechanics (largely irrelevant for API consumers) but not the what/why (definitely relevant for API consumers).

How about something like this:

> Construct a name that is suitable for replacements/whitelisting/opaque-ness look ups.

And for the namespaces method:

> Construct the name without the item's containing C++ namespaces mangled into it. In other words, the item's name within the item's namespace.